### PR TITLE
Add permissions backend SQL injections tests.

### LINKED ISSUE #1070: Add a unit test to assert #1061 has been fixed.

### PR DESCRIPTION
**Pull request by Natim**
Fixes #1070

- [x] Add tests.

### CODE REVIEW

File: tests/test_views_objects_permissions.py [hunk 1]
```diff
@@ -103,6 +103,16 @@ def test_cannot_write_if_not_allowed(self):
                           headers=self.alice_headers,
                           status=403)
 
+    def test_permission_backend_prevent_sql_injections(self):
+        self.app.get("/buckets/beer'", headers=self.headers, status=403)
+        self.app.get("/buckets/beer'/collections/barley", headers=self.headers, status=403)
+        self.app.get("/buckets/beer'/groups/barley", headers=self.headers, status=403)
+
+        self.app.get("/buckets/beer/collections/barley'", headers=self.headers, status=400)
+        self.app.get("/buckets/beer/collections/barley'/records", headers=self.headers, status=404)
```
Comment (leplatrem) [inline tests/test_views_objects_permissions.py]: interesting difference that would a deserve a comment or XXX : )

Comment (Natim) [inline tests/test_views_objects_permissions.py]: Yes it looks like a bug to me actually.

Comment (Natim) [inline tests/test_views_objects_permissions.py]: Oh no its not a bug because we have the permissions on the beer bucket so it makes sense.

Comment (leplatrem) [inline tests/test_views_objects_permissions.py]: We can add a comment for mental health of the reader

Comment (Natim) [inline tests/test_views_objects_permissions.py]: Oh no you are right we should probably say that the ID is wrong even if we are on the collection record endpoint? I don't know.

**Pull request merged**